python/hail/keytable.py

```diff
@@ -1379,5 +1379,47 @@ def range(cls, n, num_partitions=None):
 
         return KeyTable(Env.hc(), Env.hail().keytable.KeyTable.range(Env.hc()._jhc, n, joption(num_partitions)))
 
+    @handle_py4j
+    @record_method
+    @typecheck_method(i=strlike,j=strlike)
+    def maximal_independent_set(self, i, j):
+        """Compute a `maximal independent set
+        <https://en.wikipedia.org/wiki/Maximal_independent_set>__` of vertices
+        in an undirected graph whose edges are given by this KeyTable
+
+        **Examples**
+
+        Prune a minimal number of individuals from a dataset to remove all
+        relationships that are close with respect to a PC-Relate measure of
+        kinship.
+
+        >>> related_pairs = hc.import_vcf("data/sample.vcf.bgz").pc_relate(2, 0.001).filter("kin > 0.125")
+        >>> related_samples = high_kin.query('i.flatMap(i => [i,j]).collectAsSet()')
+        >>> related_samples_to_keep = high_kin.maximal_independent_set("i", "j")
+        >>> related_samples_to_remove = high_kin_samples - set(independent_set_list)
+        >>> vds.filter_samples_list(list(samples_to_remove))
+
+        **Notes**
+
+        Each row of the :class:`.KeyTable` corresponds to an undirected edge
+        between the nodes given by ``i`` and ``j``. An undirected edge may
+        appear multiple times in the :class:`.KeyTable`. It will not affect the
+        output. The node set of the graph is implicitly all the possible values
```

Comment:
The vertex set

[or use node everywhere, but wikipedia used vertex and you switch back and forth]